packages/@aws-cdk-testing/framework-integ/test/aws-apigateway/test/integ.restapi.import-deploymentstage.ts

```diff
@@ -0,0 +1,23 @@
+import * as cdk from 'aws-cdk-lib/core';
+import * as apigateway from 'aws-cdk-lib/aws-apigateway';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'integtest-restapi-import-deployment-stage');
+
+const api = new apigateway.RestApi(stack, 'my-api', {
+  deploy: false,
+});
+api.root.addMethod('GET');
+
+// Create a new deployment that uses existing stage
+new apigateway.Deployment(stack, 'MyManualDeployment', {
+  api: api,
+  stageName: 'myStage',
+});
```

Comment:
I tried to run this integ test and found the below finding. Overall great work and I'm able to create deployment for existing stage with one minor addition to the integ test as below.

1. If we manually define the deployment with `stageName` included, the deployment creates the stage with the stageName provided and creates the deployment. If the stage already exists, then it deploys to the existing stage. Hence I would suggest to add to `README.md` and also in the `stageName` property docstring that something like stage will be created if it doesn't exist.

2. Also, When the restApi model has changes, the deployment will remain the same and doesn't reflect the new changes to the existing stage. In order for the deployment to reflect the new changes to the existing stage, the user also needs to invoke the `addToLogicalId(data)` method which generates a new logical Id. For more info see [here](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.Deployment.html). Without invoking this method, the deployment will not reflect new changes in the stage.

Hence, it would be worth to mention to use the method in our integ test and readme examples if deployment is manually defined using `Deployment` construct. But for `deploy:true`,  restApi latestDeployment takes care to generate new logical Id automatically and deploys to the existing stage.

```
const api = new apigateway.RestApi(stack, 'my-api', {
  deploy: false,
});
api.root.addMethod('GET');
api.root.addMethod('POST');

const deployment = new apigateway.Deployment(stack, 'MyManualDeployment', {
  api: api,
  stageName: 'myStage',
});

deployment.addToLogicalId(`Deployment-${Date.now()}`);

new integ.IntegTest(app, 'restapi-import-deployment-stage', {
  testCases: [stack],
});
```

Apart from that, it LGTM.